internal/controller/warehouses/images.go

```diff
@@ -147,12 +156,18 @@ func imageSelectorForSubscription(
 	sub kargoapi.ImageSubscription,
 	creds *image.Credentials,
 ) (image.Selector, error) {
+	var constraint string
+	if sub.Constraint != "" {
+		constraint = sub.Constraint
+	} else if sub.ImageSelectionStrategy == kargoapi.ImageSelectionStrategySemVer && sub.SemverConstraint != "" {
+		constraint = sub.SemverConstraint
+	}
```

Comment:
```suggestion
	constraint := sub.Constraint
	if constraint == "" && sub.ImageSelectionStrategy == kargoapi.ImageSelectionStrategySemVer {
		constraint = sub.SemverConstraint
	}
```

In Go, there is a common pattern of "assign default, then override if needed".

By following this pattern, you simplify the logic by reducing from 7 lines to 4 lines while ordering the conditions more intuitively by checking if the primary value is missing first, then checking fallback conditions.

This makes it easier for developers to understand (at a later moment) because it follows the natural thought process of "do I have what I need, and if not, what are my alternatives?"